sdk/core/src/services/MetricsService.ts

```diff
@@ -1,16 +1,64 @@
 import { profile } from '@affinidi/common'
-import { metrics, EventComponent } from '@affinidi/affinity-metrics-lib'
+import { metrics, EventComponent, VcMetadata } from '@affinidi/affinity-metrics-lib'
 
-import { MetricsEvent, MetricsServiceOptions } from '../dto/shared.dto'
+import { MetricsEvent, MetricsServiceOptions, SignedCredential } from '../dto/shared.dto'
+
+type CommonVcMetadata = Omit<VcMetadata, 'data'> // anything to data will be overwritten by SpecificVcMetadada.data
+type SpecificVcMetadada = { data: any }
+
+class VcMetadataParser {
+  // parse vcType-agnostic metadata
+  private parseCommon(credential: SignedCredential): CommonVcMetadata {
+    const metadata = { vcType: credential.type }
+    return metadata
+  }
+
+  // parse vcType-specific metadata
+  /* eslint-disable-next-line no-unused-vars */
+  parseSpecific(credential: SignedCredential): SpecificVcMetadada {
+    return { data: {} }
+  }
+
+  parse(credential: SignedCredential): VcMetadata {
+    const baseMetadata = this.parseCommon(credential)
+    const addOnMetadata = this.parseSpecific(credential)
+    return { ...baseMetadata, ...addOnMetadata }
+  }
+}
+
+// TODO: move to a single file for all vcType-specific parsers?
```

Comment:
I agree, this might be a good idea to extract filters to separate files.

Reply:
Moved to a subdir `parsers` under the `services` dir as a separate module for metadata parser.